src/main/java/io/cdap/plugin/ariba/source/AribaInputSplit.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.ariba.source;
+
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+/**
+ * Ariba InputSlipt class for creating splits
+ */
+public class AribaInputSplit extends InputSplit implements Writable {
+
+  private String fileName;
+  private String jobId;
+
+  public AribaInputSplit() {
```

Comment:
why do we need this constructor?

Reply:
This is required by framework, as when I was not adding this constructor it was throwing "java.lang.RuntimeException: java.lang.NoSuchMethodException: io.cdap.plugin.ariba.source.AribaInputSplit.<init>()" exception.